docs/usage/with-typescript.md

```diff
@@ -0,0 +1,224 @@
+---
+id: with-typescript
+title: Usage With TypeScript
+sidebar_label: Usage With TypeScript
+hide_title: true
+---
+
+# Usage With TypeScript
+
+It is one of Redux Toolkit's design targets to offer great integration with TypeScript.
+
+This page is intended to give an overview of all common usecases and the most probable pitfalls you might encounter when using RTK with TypeScript.
+
+**If you encounter any problems with the Types that are not described on this page, please open an Issue!**
+
+## Using configureStore with TypeScript
+
+Using [configureStore](../api/configureStore.ms) should not need any additional typings. You might, however, want to extract the `RootState` type and the `Dispatch` type.
+
+### Getting the `State` type
+
+The easiest way of getting the `State` type is to define the root reducer in advance and extract it's `ReturnType`.  
+It's recommend to give it a different name like `RootState` to prevent confusion, as the type name `State` is usually overused.
```

Comment:
```suggestion
It is recommended to give the type a different name like `RootState` to prevent confusion, as the type name `State` is usually overused.
```